src/modules/scraper/adapter/messaging/RectifyStickyReferralConsumer.ts

```diff
@@ -0,0 +1,47 @@
+import { OnQueueFailed, Process, Processor } from "@nestjs/bull";
+import { Logger } from "@nestjs/common";
+import { Job } from "bull";
+import { IsNull, MoreThan, Not, Repository } from "typeorm";
+import { InjectRepository } from "@nestjs/typeorm";
+
+import { DepositReferralQueueMessage, RectifyStickyReferralQueueMessage, ScraperQueue } from ".";
+import { Deposit } from "../../../deposit/model/deposit.entity";
+import { ScraperQueuesService } from "../../service/ScraperQueuesService";
+
+@Processor(ScraperQueue.RectifyStickyReferral)
+export class RectifyStickyReferralConsumer {
+  private logger = new Logger(RectifyStickyReferralConsumer.name);
+
+  constructor(
+    @InjectRepository(Deposit) readonly depositRepository: Repository<Deposit>,
+    private scraperQueuesService: ScraperQueuesService,
+  ) {}
+
+  @Process({ concurrency: 1 })
+  private async process(job: Job<RectifyStickyReferralQueueMessage>) {
+    const { depositId } = job.data;
+    const deposit = await this.depositRepository.findOne({ where: { id: depositId } });
+
+    if (!deposit) return;
+
+    // After extracting the referral address, all deposits with later deposit time must have
+    // the sticky referral address updated
+    const deposits = await this.depositRepository.find({
+      where: {
+        depositorAddr: deposit.depositorAddr,
+        depositDate: MoreThan(deposit.depositDate),
+      },
+    });
```

Comment:
Do we need to sort these by the deposit date or doesn't the order matter here?

Reply:
Order doesn't matter here, but I added sorting by deposit date because it makes from a logical point of view. Thank you!